ast/node.go

```diff
@@ -522,3 +525,56 @@ type LambdaExpr struct {
 }
 
 func (i *LambdaExpr) Pos() Pos { return i.Lcurlybrace }
+
+// stmtNode() ensures that only ExComamnd and Comment nodes can be assigned to
+// an Statement.
+//
+func (*Break) stmtNode()      {}
+func (*Catch) stmtNode()      {}
+func (*Continue) stmtNode()   {}
+func (DelFunction) stmtNode() {}
+func (*EchoCmd) stmtNode()    {}
+func (*Echohl) stmtNode()     {}
+func (*Else) stmtNode()       {}
+func (*ElseIf) stmtNode()     {}
+func (*EndFor) stmtNode()     {}
+func (EndFunction) stmtNode() {}
+func (*EndIf) stmtNode()      {}
+func (*EndTry) stmtNode()     {}
+func (*EndWhile) stmtNode()   {}
+func (*ExCall) stmtNode()     {}
+func (Excmd) stmtNode()       {}
+func (*Execute) stmtNode()    {}
+func (*Finally) stmtNode()    {}
+func (*For) stmtNode()        {}
+func (Function) stmtNode()    {}
+func (*If) stmtNode()         {}
+func (*Let) stmtNode()        {}
+func (*LockVar) stmtNode()    {}
+func (*Return) stmtNode()     {}
+func (*Throw) stmtNode()      {}
+func (*Try) stmtNode()        {}
+func (*UnLet) stmtNode()      {}
+func (*UnLockVar) stmtNode()  {}
+func (*While) stmtNode()      {}
+
+func (*Comment) stmtNode() {}
+
+// exprNode() ensures that only expression nodes can be assigned to an Expr.
+//
+func (*TernaryExpr) exprNode()   {}
+func (*BinaryExpr) exprNode()    {}
+func (*UnaryExpr) exprNode()     {}
+func (*SubscriptExpr) exprNode() {}
+func (*SliceExpr) exprNode()     {}
+func (*CallExpr) exprNode()      {}
+func (*DotExpr) exprNode()       {}
+func (*BasicLit) exprNode()      {}
+func (*List) exprNode()          {}
+func (*Dict) exprNode()          {}
+func (*KeyValue) exprNode()      {}
```

Comment:
<sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
func exprNode is unused (U1000)